File: phi_plugin/model/type/SongsInfo.ts
import { idString, songString } from "./type"
import Chart from "../class/Chart"


export default interface SongsInfo {
    id: idString
    song: songString
    illustration: string
    illustration_big: string
    can_t_be_letter: boolean
    can_t_be_guessill: boolean
    chapter: string
    bpm: string
    composer: string
    length: string
    illustrator: string
    spinfo: string
    chart: { [key: string]: Chart }
}

File: phi_plugin/model/cls/SongsInfo.py
from pydantic import BaseModel

from ..cls.Chart import ChartModel
from .type import idString, songString


class SongsInfoModel(BaseModel):
    id: idString
    song: songString
    illustration: str
    "曲绘略缩图"
    illustration_big: str
    "曲绘"
    can_t_be_letter: bool
    can_t_be_guessill: bool
    chapter: str
    bpm: str
    composer: str
    length: str
    illustrator: str
    spinfo: str
    chart: dict[str, ChartModel]
